plugins/modules/filesystem.py

```diff
@@ -197,6 +198,121 @@ def fs_state(module, filesystem):
     return False
 
 
+def check_attr_change(module, filesystem):
+    """
+    Determines if changes will be made on the filesystem.
+    param module: Ansible module argument spec.
+    param filesystem: filesystem name.
+    return: True - changes will be made on the filesystem / False = filesystem will remain unchanged
+    """
+
+    cmd = "lsfs -cq %s" % filesystem
+    rc, stdout, stderr = module.run_command(cmd)
+    all_attr = stdout.splitlines()
+    old_attr = all_attr[1].split(":")
+
+    # check for extended attributes
+    old_ext_attr = dict()
+    if len(all_attr) == 3:
+        attrs = re.sub(r"[()]", "", all_attr[2]).strip()
+        attrs = attrs.split(":")
+        for attr in attrs:
+            attr = attr.rsplit(" ", 1)
+            key = re.sub(" ", "_", attr[0]).lower()
+            old_ext_attr[key] = attr[1]
+
+    new_attr = dict()
+    attrs = module.params["attributes"]
+    if attrs:
+        for attr in attrs:
+            attr = attr.strip()
+            attr = attr.split("=")
+            new_attr[attr[0]] = attr[1]
+
+    # check if mount group changed
+    old_mnt_grp = old_attr[4]
```

Comment:
Does this current algorithm relies on the old keys being allocated in the precise location? ( old_attr[4] ). 
 Should we key it by the name of the attributes that we know as we are comparing it with specific new attributes?